expr/func_divideserieslists.go

```diff
@@ -0,0 +1,75 @@
+package expr
+
+import (
+	"errors"
+	"fmt"
+	"math"
+
+	"github.com/grafana/metrictank/api/models"
+	"gopkg.in/raintank/schema.v1"
+)
+
+type FuncDivideSeriesLists struct {
+	dividends GraphiteFunc
+	divisors  GraphiteFunc
+}
+
+func NewDivideSeriesLists() GraphiteFunc {
+	return &FuncDivideSeriesLists{}
+}
+
+func (s *FuncDivideSeriesLists) Signature() ([]Arg, []Arg) {
+	return []Arg{
+		ArgSeriesList{val: &s.dividends},
+		ArgSeriesList{val: &s.divisors},
+	}, []Arg{ArgSeries{}}
+}
+
+func (s *FuncDivideSeriesLists) Context(context Context) Context {
+	return context
+}
+
+func (s *FuncDivideSeriesLists) Exec(cache map[Req][]models.Series) ([]models.Series, error) {
+	dividends, err := s.dividends.Exec(cache)
+	if err != nil {
+		return nil, err
+	}
+	divisors, err := s.divisors.Exec(cache)
+	if err != nil {
+		return nil, err
+	}
+	if len(divisors) != len(dividends) {
+		return nil, errors.New("dividendSeriesList and divisorSeriesList argument must have equal length")
+	}
+
+	var series []models.Series
+	for i, dividend := range dividends {
+		divisor := divisors[i]
+		out := pointSlicePool.Get().([]schema.Point)
+		for i := 0; i < len(dividend.Datapoints); i++ {
+			p := schema.Point{
+				Ts: dividend.Datapoints[i].Ts,
+			}
+			if divisor.Datapoints[i].Val == 0 {
+				p.Val = math.NaN()
+			} else {
+				p.Val = dividend.Datapoints[i].Val / divisor.Datapoints[i].Val
```

Comment:
This doesn't seem equivalent to safeDiv. What about dividing by 0

Reply:
golang handles that : https://play.golang.org/p/lpkONXNAkcq

As far as I can google, this isn't well-defined (this implementation was inspired from `divideSeries`).

Reply:
Ahh I was trying with ints :) forgot it's all floats
NaN is diff from Inf though so it'll be slightly different